crates/miden-objects/src/account/builder/mod.rs

```diff
@@ -100,15 +120,22 @@ impl AccountBuilder {
         #[cfg(all(not(feature = "testing"), not(test)))]
         let vault = AssetVault::default();
 
-        let (code, storage) =
-            Account::initialize_from_components(self.account_type, &self.components).map_err(
-                |err| {
-                    AccountError::BuildError(
-                        "account components failed to build".into(),
-                        Some(Box::new(err)),
-                    )
-                },
-            )?;
+        let auth_component = self
+            .auth_component
+            .clone()
+            .ok_or(AccountError::BuildError("auth component must be set".into(), None))?;
+
+        // Make sure the auth component is first.
+        let mut components = vec![auth_component];
+        components.extend_from_slice(&self.components);
```

Comment:
This will create a clone of all components. Maybe we can do this instead to avoid the cloning:

```rust
let mut components = core::mem::take(&mut self.components);
components.push(auth_component);
let auth_component_index = components.len() - 1;
// Swap the auth component to index 0.
components.swap(0, auth_component_index);
```

Reply:
Done in [40b9b45](https://github.com/0xMiden/miden-base/pull/1480/commits/40b9b458b61965259ddde9afb060f76659c4c75a)